spark/v3.2/spark/src/test/java/org/apache/iceberg/spark/TestSparkSessionCatalog.java

```diff
@@ -86,4 +97,23 @@ public void testValidateHmsUri() {
             .defaultNamespace()[0]
             .equals("default"));
   }
+
+  @Test
+  public void testLoadFunction() {
+    spark.sessionState().catalogManager().reset();
+    spark.conf().set(envHmsUriKey, hmsUri);
+    spark.conf().set(catalogHmsUriKey, hmsUri);
+
+    String functionClass = "org.apache.hadoop.hive.ql.udf.generic.GenericUDFUpper";
+
+    // load permanent UDF in Hive via FunctionCatalog
+    spark.sql(String.format("CREATE FUNCTION upper AS '%s'", functionClass));
```

Comment:
Spark has a build-in function also named `upper`. I think creating a existing function will not invoke `SparkSessionCatalog#loadFunction`, can you confirm that?

Reply:
Let me change the function name used here.